src/app-layout/__tests__/trigger-button.test.tsx

```diff
@@ -0,0 +1,357 @@
+} from '../../../lib/components/app-layout/visual-refresh-toolbar/toolbar/trigger-button/index.js';
+import * as AllContext from '../../../lib/components/app-layout/visual-refresh/context.js';
+import visualRefreshStyles from '../../../lib/components/app-layout/visual-refresh/styles.css.js';
+import toolbarTriggerButtonStyles from '../../../lib/components/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.css.js';
+
+import { act, render } from '@testing-library/react';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+const MockUseAppLayoutInternals = jest.spyOn(AllContext, 'useAppLayoutInternals').mockName('useAppLayoutInternals');
+
+const mockDrawerId = 'mock-drawer-id';
+const mockTestId = `awsui-app-layout-trigger-${mockDrawerId}`;
+
+const mockProps = {
+  ariaLabel: 'Aria label',
+  className: 'class-from-props',
+  iconName: 'bug',
+  testId: mockTestId,
+};
+const mockOtherEl = {
+  class: 'other-el-class',
+  text: 'other-element',
+};
+const mockUseLayoutInternalValues = {
+  hasOpenDrawer: false,
+  isMobile: false,
+};
+
+const mockEventBubble = {
+  bubbles: true,
+  isTrusted: true,
+  relatedTarget: null,
+};
+
+const mockEventBubbleWithShiftFocus = {
+  ...mockEventBubble,
+  relatedTarget: {
+    dataset: {
+      shiftFocus: 'last-opened-toolbar-trigger-button',
+    },
+  },
+};
+
+const delay = () => act(() => new Promise(resolve => setTimeout(resolve)));
+
+const renderVFTriggerButton = async (
+  props: Partial<VFTriggerButtonProps> = {},
+  useAppLayoutInternalsValues: Partial<AllContext.AppLayoutInternals> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  MockUseAppLayoutInternals.mockReturnValue({
+    ...mockUseLayoutInternalValues,
+    ...useAppLayoutInternalsValues,
+  } as AllContext.AppLayoutInternals);
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VFTriggerButton {...(renderProps as VFTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(visualRefreshStyles['trigger-wrapper'])!;
+  await delay();
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+const renderVFTBTriggerButton = async (
+  props: Partial<VFTBTriggerButtonProps> = {},
+  ref: React.Ref<ButtonProps.Ref>
+) => {
+  const renderProps = { ...mockProps, ...props };
+  const { container, rerender, getByTestId, getByText } = render(
+    <div>
+      <VFTBTriggerButton {...(renderProps as VFTBTriggerButtonProps)} ref={ref} />
+      <span className={mockOtherEl.class}>{mockOtherEl.text}</span>
+    </div>
+  );
+  const wrapper = createWrapper(container).findByClassName(toolbarTriggerButtonStyles['trigger-wrapper'])!;
+  await delay();
+  return { wrapper, rerender, getByTestId, getByText };
+};
+
+describe('Visual refresh trigger-button (not in appLayoutWidget toolbar)', () => {
```

Comment:
What are the advantages of testing these buttons separately, considering they are not self-sufficient components but rather part of layouts?

Reply:
Isolation of the component, knowing it works as expected in isolation so easier to pinpoint failures from the parent. Also keeps coverage high.  This will be needed later as we add the tooltip functionality to the trigger button, we want to make sure the tooltip is working correctly insolation.